src/app/application.cpp

```diff
@@ -146,6 +146,13 @@ Application::Application(const QString &id, int &argc, char **argv)
     Logger::instance()->addMessage(tr("qBittorrent %1 started", "qBittorrent v3.2.0alpha started").arg(QBT_VERSION));
 }
 
+Application::~Application()
+{
+    // we still need to call cleanup()
+    // in case the App failed to start
+    cleanup();
```

Comment:
I lost focus on the topic, can someone explain why we need this now?

Reply:
Now the `Application::exec()` can be interrupted before event loop is started in case of error (WebUI failed to start, e.g.). We need to correctly cleanup already initialized components.

Reply:
RAII?

Reply:
>We need to correctly cleanup already initialized components.

Thank you, I understand now.